crates/astria-conductor/src/executor/mod.rs

```diff
@@ -226,13 +273,45 @@ impl Executor {
         // XXX: shut down the channels here and attempt to drain them before returning.
     }
 
-    fn is_spread_too_large(&self) -> bool {
+    /// Calculates the maximum allowed spread between firm and soft commitments heights.
+    ///
+    /// The maximum allowed spread is taken as `max_spread = variance * 6`, where `variance`
+    /// is the `celestia_block_variance` as defined in the rollup node's genesis that this
+    /// executor/conductor talks to.
+    ///
+    /// The heuristic 6 is the largest number of Sequencer heights that will be found at
+    /// one Celestia height.
+    ///
+    /// # Panics
+    /// Panics if the `u32` underlying the celestia block variance tracked in the state could
+    /// not be converted to a `usize`. This should never happen on any reasonable architecture
+    /// that Conductor will run on.
+    fn calculate_max_spread(&self) -> usize {
+        usize::try_from(self.state.borrow().celestia_block_variance())
+            .expect("converting a u32 to usize should work on any architecture conductor runs on")
+            .saturating_mul(6)
+    }
+
+    /// Returns if the spread between firm and soft commitment heights in the tracked state is too
+    /// large.
+    ///
+    /// Always returns `false` if this executor was configured with `consider_commitment_spread =
+    /// false`.
+    fn is_spread_too_large(&self, max_spread: usize) -> bool {
         if !self.consider_commitment_spread {
             return false;
         }
-        let next_firm = self.state.borrow().next_firm_sequencer_height().value();
-        let next_soft = self.state.borrow().next_soft_sequencer_height().value();
-        let is_too_far_ahead = next_soft.saturating_sub(next_firm) >= 16;
+        let (next_firm, next_soft) = {
+            let state = self.state.borrow();
+            let next_firm = state.next_firm_sequencer_height().value();
+            let next_soft = state.next_soft_sequencer_height().value();
+            (next_firm, next_soft)
+        };
+
+        let is_too_far_ahead = usize::try_from(next_soft.saturating_sub(next_firm))
```

Comment:
In what case would this `try_from` not be successful? It seems like this would be a larger issue?

Reply:
If you ran this on a 16bit architecture (or just sub-32bit architecture) this would fail. I think it's fair to assume that we are running on at least 32 bit (and even that is unrealistic nowadays) so that there is an `expect` right after.